### PR TITLE
Implement Session GC Cli

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue # .

### Summary of Changes
Since Joomla 3.8.4, if you use Database Session Handler, session data won't be deleted on page loads like in older version of Joomla anymore, it now relies on PHP session data garbage collection proces to delete session data and could cause session data stay longer in database than expected. On some hostings, session.gc_probability http://php.net/manual/en/session.configuration.php#ini.session.gc-probability even has value set to 0, mean session data won't be deleted at all.

This PR implements a simple CLI script so that you can setup cron job to clean up session data if needed. It will be more reliable than rely on session data garbage collection and for high traffic website, you will get big performance improvement (set session.gc_probability  to 0 and setup cron job to delete session data- see https://github.com/joomla/joomla-cms/issues/19146 for a use case)

### Testing Instructions
1. Go to System - Global Configuration, look at System tab, set Session Handler to Database, Session Lifetime to 1 minute

2. Access to frontend of your site so that a new record will be inserted into #__session table

3. Access to your site database via PHPMyadmin, look at #__session, you should see some records

4. Wait for about one minute

5. Run this CLI script. I don't know about Linux, but on Windows, I just need to run this command from Command Line **php D:\www\joomla\cli\sessionGc.php**  (D:\www\joomla needs to be replaced with the path of your Joomla installation)

6. Check #__session table again, you should see that the old records were deleted (maybe all records deleted)

### Documentation Changes Required
Maybe we need to document this new feature if accepted

### CODE REVIEW

**Comment by csthomas** [comment]
Another way can be only:
```php
	public function doExecute()
	{
		$config = JFactory::getConfig();

		$expire = $config->get('lifetime');
		$expire = $expire ? $expire * 60 : 900;

		$handler = $config->get('session_handler', 'none');
		$options = array('expire' => $expire);

		$storage = JSessionStorage::getInstance($handler, $options);
		$storage->gc($expire);
	}
```

**Comment by mbabker** [comment]
Honestly, direct calling the storage layer of the session should be avoided, that part of the API is typically an internal detail and in 4.0 you honestly don't have direct access to it (you have to go through a `Joomla\Session\SessionInterface` implementation, in the CMS that's `Joomla\CMS\Session\Session` to interact with sessions at any level unless you're using Reflection to get access to stuff you shouldn't).  So what is there now is fine.

File: cli/sessionGc.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function doExecute()
+	{
+		JFactory::getSession()->gc();
+	}
+}
+
+JApplicationCli::getInstance('SessionGc')->execute();
```
Comment (csthomas) [inline cli/sessionGc.php]: PR works, please add a new line.

Comment (joomdonation) [reply cli/sessionGc.php]: Done, thanks !

**Comment by ghost** [comment]
@csthomas can you please mark your Test as successfully?

**Comment by csthomas** [comment]
I have tested this item :white_check_mark: successfully on bb1bb9374a9721fdfbddb116af43af0e87afe365<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19548">issues.joomla.org/tracker/joomla-cms/19548</a>.</sub>

**Comment by joomdonation** [comment]
@mbabker Can this one go into 3.8.5? It could be useful for site using Database Handler running on host has session.gc-probability set to 0, they can use this core cli instead of writing the code themself

**Comment by brianteeman** [comment]
It's a new feature so semantic versioning says 3.9

**Comment by joomdonation** [comment]
@brianteeman It is an additional feature to provide a way to _fix_ the change we made in 3.8.4, so It could go into 3.8.5 if it is possible

**Comment by mbabker** [comment]
3.8.5 has a specific purpose and is not getting crammed full of miscellaneous stuff.

**Comment by mbabker** [comment]
No longer required pending acceptance of #19678

**Comment by mbabker** [comment]
Changed my mind.  Merging this.  We need a plugin to do session gc as pointed out in #19585 and this PR adds the API to do it.  SemVer be damned right now I guess.

**Pull request merged**